tests/Unit/CallbackRegistrarTest.php

```diff
@@ -85,41 +88,78 @@ public function testIfParseCallbacksLoadsCallbacksFromClass()
 
     public function testIfCallbackReturnsCallable()
     {
-        $registrar = new CallbackRegistrar;
-        $registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
+        $this->registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
 
-        $this->assertEquals('\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo', $registrar->callback('foo'));
-        $this->assertEquals('\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo', $registrar->callback('orFoo'));
+        $this->assertEquals('\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo', $this->registrar->callback('foo'));
+        $this->assertEquals('\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo', $this->registrar->callback('orFoo'));
     }
 
     public function testIfHasCallbackReturnsTrueIfCallbackExists()
     {
-        $registrar = new CallbackRegistrar;
-        $registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
+        $this->registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
 
-        $this->assertTrue($registrar->hasCallback('foo'));
+        $this->assertTrue($this->registrar->hasCallback('foo'));
     }
 
     public function testIfHasCallbackReturnsFalseIfCallbackDoesNotExist()
     {
-        $registrar = new CallbackRegistrar;
-
-        $this->assertFalse($registrar->hasCallback('foo'));
+        $this->assertFalse($this->registrar->hasCallback('foo'));
     }
 
     public function testIfHasProxyReturnsTrueIfProxyExists()
     {
-        $registrar = new CallbackRegistrar;
-        $registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
+        $this->registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
 
-        $this->assertTrue($registrar->hasProxy('orFoo'));
+        $this->assertTrue($this->registrar->hasProxy('orFoo'));
     }
 
     public function testIfHasProxyReturnsFalseIfProxyDoesNotExist()
     {
-        $registrar = new CallbackRegistrar;
+        $this->assertFalse($this->registrar->hasProxy('orFoo'));
+    }
+
+    public function testIfSetDefaultInvokesCallbackIfArgIsString()
+    {
+        $this->registrar = Mockery::mock(CallbackRegistrar::class)->makePartial();
+        $this->registrar->loadCallbacks(['foo' => '\LaraCrafts\GeoRoutes\Tests\Mocks\Callbacks::foo']);
+
+        $this->registrar->shouldReceive('callback')->with('foo')->once();
+
+        $this->registrar->setDefault('foo');
+    }
```

Comment:
Omitting this test means that we'll have non-test covered code, also this test assures that the method does not interpret callables (e.g. FooClass::bar) as strings which would result in an exception.

Reply:
In my opinion you should test `setDefault` with `getDefault`. As I said test the functionality not the implementation.

Reply:
I have thought of that during the testing phase, but that would make the tests tightly coupled and would cause both to fail if the `getDefault` method is failing. In addition, in my own humble opinion, this is actually testing the functionality which is accepting both callables and strings and in return setting the default value.

Reply:
To be exact my problem is at line 126. With that you are testing the implementation. However, as far as I know you made `setDefault` to return something. in that case you can test the return of that function. Lastly, getters and setters are usually "tightly coupled" similarly to push and pop.